rules/type-declaration/tests/Rector/Identical/FlipTypeControlToUseExclusiveTypeRector/Fixture/fixture.php.inc

```diff
@@ -0,0 +1,51 @@
+<?php
+
+namespace Rector\TypeDeclaration\Tests\Rector\Identical\FlipTypeControlToUseExclusiveTypeRector\Fixture;
+
+use stdClass;
+
+class Fixture
+{
+    public function run()
+    {
+        /** @var stdClass|null $stdClass */
+        $stdClass = $this->getStdClass();
+        if ($stdClass === null) {
+            return;
+        }
+
+        /** @var null|stdClass $stdClass */
+        $stdClass = $this->getStdClass();
+        if ($stdClass === null) {
+            return;
+        }
```

Comment:
Oh, I see :). I probably described incorrectly.

It's not about flipping the type in docs. It's rather about flipping the condition of comparison.

Why? In this case, if the code continues, the `$stdClass` is anything but `null`:

```php
 if ($stdClass === null) {
    return;
 }

 // what is stdClass exactly?
```

But in this case, we know it's exactly one type - `SomeType`:

```php
 if (! $stdClass instanceof SomeType) {
    return;
 }

 // we know $stdClass is SomeType
```

So the idea is to converted `=== null` comparison to `instanceof` checks.

```diff
-if ($stdClass === null) {
+if (! $stdClass instanceof SomeType) {
    return;
 }
```

That way code is more robus and we don't have to relly on doblocks above. So they can be droped too.


```diff
-/** @var SomeType|null $stdClass */
-if ($stdClass === null) {
+if (! $stdClass instanceof SomeType) {
    return;
 }
```

Let me know if you need more examples.

Reply:
implemented.

Reply:
if it not rely on types in the doc, the following should seems still need to be skipped as in the fixture:

```php
        /** @var stdClass|null|Foo|Bar $stdClass */
        $stdClass = $this->getStdClass();
        if ($stdClass === null) {
            return;
        }
 ```

Reply:
> the following should seems still need to be skipped as in the fixture:

Agreed :+1: 

It should keep only 1 type + `null`

Reply:
> the following should seems still need to be skipped as in the fixture:

Agreed :+1: 

It should keep only 1 type + `null`, other should skip